### ISSUE TITLE
`DEBIAN_FRONTEND=noninteractive` not working when installing apt-get freeipa-server

### ISSUE THREAD

**Issue by yabhinav** [pypa/virtualenv#1022]
For reference, look at duplicate Issue on ansible GitHub:
https://github.com/ansible/ansible/issues/21691

SUMMARY

When installing freeipa-server the post installation script try to setup the IPA server or krb5 configuration.
To prevent that apt-get install should be run with DEBIAN_FRONTEND=noninteractive apt-get update && apt-get install -yf freeipa-server

```shell
"export DEBIAN_FRONTEND=noninteractive &&  apt-get install -yf freeipa-server dnsutils sssd-ldap expect "
"DEBIAN_FRONTEND=noninteractive  apt-get install -yf freeipa-server dnsutils sssd-ldap expect "
```

Installing freeipa-client and freeipa-server with command line fails when running on virutalenv /bin/bash

The issue is probably due to incompatibility with docker 1.13.1 + ubuntu 16.04 + virtualenv

```
docker exec  "$(cat ${container_id})" bash -c 'source ~/.bashrc; workon ansible_2.2.0.0; ansible --version; export DEBIAN_FRONTEND=noninteractive && apt-get update -q  && apt-get install -q -y -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" freeipa-server freeipa-client '
```

The same command works on [ubuntu:14.04](https://hub.docker.com/_/ubuntu/), while it fails on [ubuntu:16.04](https://hub.docker.com/_/ubuntu/) docker image

Also the following installation command works on ubuntu16:04 when  any python virtialenv is not activated 
```"apt-get update && apt-get -y install freeipa-server"```

So the issue is not with docket images but with the python-virtual env , which is installed wither with pip or with apt-get. They are not reading the environment variable `DEBIAN_FRONTEND=noninteractive`

**Comment by yan12125** [comment]
I guess you mean ```export DEBIAN_FRONTEND=noninteractive &&  apt-get install -yf freeipa-server dnsutils sssd-ldap expect``` does not work in virtual environments? That works fine for me: https://gist.github.com/yan12125/66a6760a255588c2ca052653b936b1ed

In your commands, ```workon``` is provided by [virtualenvwrapper](https://bitbucket.org/dhellmann/virtualenvwrapper/), not virtualenv. There were reports that things working with virtualenv are broken with virtualenvwrapper. (#1006) You may want to also report this issue there.

**Comment by yabhinav** [comment]
Hi @yan12125  it doesn't work when you are using non-interactive ansible-playbooks or scripts that run   on the docker image ``ubuntu16:04``  

I will try you idea without using virtualenvwrapper!!

**Comment by dhoppeIT** [comment]
I was able to reproduce the problem. On my MacBook (Python 2.7.15, Virtualenv 16.0.0) everything works fine, but on a CentOS instance (Python 2.7.5, Virtualenv 15.1.0) I cannot install packages in a Docker container (Debian 8, Ubuntu 16.04) because the apt-utils package is missing. 

So I uninstalled the python-virtualenv package on the CentOS instances and installed Virtualenv 16.0.0 via pip. I also compiled Python 2.7.15 and created a new Virtualenv, based on that version but the problem still occurs
